PilotDataPlatform/metadata
# PILOT-1700: Update kafka items test cases and docker image

## Summary

- During local execution of test cases that rely on Kafka, some test cases would periodically fail during development due to incorrect offset seeking and consumer closure. This is now fixed by ensuring Kafka consumer is only initialized once during the tests, and seeking to last offset, in order to read most recent message.
- Docker image updated to confluent (**apache 2 licence**). Previous image periodically stopped broker mid-way through tests.

## JIRA Issues

- https://indocconsortium.atlassian.net/browse/PILOT-1700

## Type of Change


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Spin up docker compose locally.
2. Run test cases in `test_items.py `. All cases should pass locally.
